### PR TITLE
Fixed: As Many UI Issues as I can Handle

### LINKED ISSUE #1551: Some SVG Path Icons are a bit too big with respect to Figma
We noticed that just coping the Geometry data appears to not always result in icons of the correct size.

Resizing the resulting `UnifiedIcon` with Size property does scale the icon, but the starting size is incorrect so all scaling ends up being wrong.

Solution seems to be to have the entire SVG data, not just the Geometry data. That appears to result in correct size.

### LINKED ISSUE #1658: My Mods list - Toggles keep being pushed off screen.
# Bug Report

## Summary
The toggles on the right side of the mods table seem to keep "falling" out of the view.

## Steps to reproduce
View the mods screen in a window. 

## What is the expected behaviour?
The columns are visible if they fit into the current view. 

## Other information

https://github.com/Nexus-Mods/NexusMods.App/assets/31670524/1b2a29a2-2f87-40a6-8b80-9b3bd9211884

### PR DESCRIPTION
**Pull request by Sewer56**
## Summary

This PR fixes as many possible discrepancies between the App UI and the Figma design as possible.

Methodology is as follows:

- Set the App window to 70% opacity.
- Set the Figma project to 100% size. (Ctrl+0)
- Overlay the App Window ontop of the Figma

If the App Window does not match Figma, figure out what's wrong with the App.

In this PR I've basically went over the entire UI of the App, and made almost everything match 1:1 pixel by pixel with Figma.

## Commit / Change List

- [Fixed: Buttons no longer have excessive padding on left and right.](https://github.com/Nexus-Mods/NexusMods.App/commit/09a3aad41dc17f561e937fc93fcfaf253854622c)
    - Reduced from 7 to 4, now icon buttons are round, and design is more closely matched.
    - Example: Icon in Pill now Makes Rounded Circle, instead of a thick rectangle.
- [Fixed: Icon sizes in buttons being too small.](https://github.com/Nexus-Mods/NexusMods.App/commit/09a3aad41dc17f561e937fc93fcfaf253854622c)
    - Icons inside buttons were 20px but should have been 24px (Design).
    - Example: AdvancedInstaller, My Games
- [Fixed: Help Icon in Top Bar now Matches Sizes of Other Icons](https://github.com/Nexus-Mods/NexusMods.App/commit/09a3aad41dc17f561e937fc93fcfaf253854622c)
- [Fixed: Description Icon was too small in FOMOD Installer](https://github.com/Nexus-Mods/NexusMods.App/commit/09a3aad41dc17f561e937fc93fcfaf253854622c)
- [Fixed: My Games icon now shows filled GamePad, like on Website](https://github.com/Nexus-Mods/NexusMods.App/commit/a320027bda1cadfc1635fb2bb8d7ff5e73cc2c66)
- [Removed: Legacy Unused Joystick 'My Games' Icon](https://github.com/Nexus-Mods/NexusMods.App/commit/63b1b8447905b10851705dc359687e1f5fe9ce3d)
- [LoadoutGrid now shows correct PlaylistRemove instead of Close Button](https://github.com/Nexus-Mods/NexusMods.App/commit/ad12a3d46ab0c108451fa76e2c996aeb9aafa449)
- [Rename: Health check -> Health Check](https://github.com/Nexus-Mods/NexusMods.App/commit/84839ce8d2215d39f039d94f208b01a267d51e50)
- [Removed: Unused Description Icon](https://github.com/Nexus-Mods/NexusMods.App/commit/43b1e741a2ab7396ebf2edefd108dc8e4e3261be)
- [Improved: Actions Section in Mod Library](https://github.com/Nexus-Mods/NexusMods.App/commit/3975779a247f05f069200b96da59ab8a386c6d94)
    - [Part 2](https://github.com/Nexus-Mods/NexusMods.App/commit/5a44c278f82b868ca6485cf7d6d8f00d36f3aea6)
    - The text no longer clips out of the button if the column is resized down.
    - The column is no longer resized down when the window is resized.
- [Fixed: Text Casing on Advanced add and View Mod files](https://github.com/Nexus-Mods/NexusMods.App/commit/1017a0b9378bad24d9ccb134e2921c6dab3fed2c)
- [Fixed: Incorrect (missing) scale on Download Button Click](https://github.com/Nexus-Mods/NexusMods.App/commit/46b831d0939b90ec7254d421b8d52f52cdd2ca7f)
    - Pressing the `Download` button now shrinks it to the same amount as the loadout and nexus icon on the left spine.
- [Fixed: Diagnostic Details Tab showing Wrong (Old) Icon](https://github.com/Nexus-Mods/NexusMods.App/commit/f3fe49bcdcfa84ed00ba917335505a6b36dfd3bb)
- [Removed Unused Legacy `DiagnosticPage` Icon and Using SVG Import To Ensure Icon Sizes Match Figma](https://github.com/Nexus-Mods/NexusMods.App/commit/cf8fd61a8268a3e77c24830b690868b6cccf7cb1)
    - [Part 2](https://github.com/Nexus-Mods/NexusMods.App/commit/3adc5b488ac8fd5764f6dd50046daa2e35f1160d)
- [Fixed: The 'State' Column Text in Diff View is now Centered](https://github.com/Nexus-Mods/NexusMods.App/commit/5449f96c6fa6768bb395e3bf4d1fabb29c053e17)
- [Changed: Tree Data Grid Default Column Order to Match Design](https://github.com/Nexus-Mods/NexusMods.App/commit/ebd85aee8950cfaa7e9c44d1590bfbb8f90574b7)
- [Fixed: Advanced Installer and FileTreeView Icon Sizes Now Match Design](https://github.com/Nexus-Mods/NexusMods.App/commit/434d2aa8eb17f16e28db2844df35fbeec7209b36)
- [Changed: File icon in FileTreeView to outline](https://github.com/Nexus-Mods/NexusMods.App/commit/6a89574de5f5877be9adb517ae999ee39dfe59a9)
    - This icon is used in FileTreeView and Nowhere else. So the entry was modified directly to match the current icon in `FileTreeView`
- [Fixed: Incorrect Link on Progress-Download Icon](https://github.com/Nexus-Mods/NexusMods.App/commit/b3b7dc48c8940b25767a0611f9e217bcfe12fbaf)
    - [Part 2](https://github.com/Nexus-Mods/NexusMods.App/commit/7591cc909161c765664114d6be969c350fe429a2)
- [Fixed: Left Menu Icons now Styled via Styles](https://github.com/Nexus-Mods/NexusMods.App/commit/3e2bafd6bed1c26404a74d07d14a181df8b46720)
    - The sizes now also match Figma, and are upscaled to 20px with confirmation from @captainsandypants 
- [Added: Instructions for Importing SVGs Correctly](https://github.com/Nexus-Mods/NexusMods.App/commit/04f2cbc94c0fb00b9474d916a71d3a6181047b74)
    - Note: This is temporary and will need further refinement. Will open a ticket for this.
    
2024 Jun 20:

- [Removed: All button scaling on press as per design request](https://github.com/Nexus-Mods/NexusMods.App/commit/481ac94d69d0b44f007ca77a02965d6e5be6f374)
- [Fixed: Tab Header Text is now Vertically Center Aligned](https://github.com/Nexus-Mods/NexusMods.App/commit/f2b243ef5e8a01c4cdca90afbcc3db98345be97e)
- [Fixed: Toolbar Item Text not Vertically Alignment, ComboBox Text Alignment & Disabled Style](https://github.com/Nexus-Mods/NexusMods.App/commit/c55a27b27a857f3a010b8e276e40867ac71fea96)

## Fixes

fixes #1551
fixes #1658 (to some degree)

And a bunch of other non-reported tickets.

### CODE REVIEW

**Comment by erri120** [comment]
Did you rebase already? Some of these fixes might already been in main.

**Comment by Sewer56** [comment]
> Some of these fixes might already been in main.

Only renaming `Health check` to `Health Check`.
It happened to be community PR'd as I was working on this branch.

Note: This branch was started around midnight today when I woke up.

**Review by Al12rs — APPROVED** [review]

**Comment by Sewer56** [comment]
Please note, phrasing like

> Changing this default has caused a bunch of icons over the app to now not have correct size.

Needs further explanation.
Did UI things change on main?
Which icons are we talking about? etc.

I spent the whole day last Thursday frantically overlaying the App window over the Designs, to ensure each icon lines up.

I wound up doing that for just about every view in the App, so if I've missed some cases e.g. 'X is selected', those cases need to be stated exactly on the ticket.

-----

Misc Note: Design decided to update LeftMenu icons to the size in this PR, but because the decision was done late Thursday, it probably hasn't been reflected in actual design files.

**Comment by Al12rs** [comment]
The issues we found in review huddle with Design were:
- Selected tab headers have different icon sizes from non selected ones
- GameWidget icons were inconsistent (pushed a commit to fix this)
- All FileTrees had increased sizes now, not sure if this actually matched designs but we might have decided to use smaller size at the time instead since it was more compact.
- Icons in various buttons were now different in size, e.g. advanced installer, GameWidget etc.

Given the various inconsistencies, one would have to go through the entire app again to check each icon.
Changing the default size like that is not advised because it requires going though the entire app, both for the implementer and the reviewer, which takes a lot of time.

**Comment by Sewer56** [comment]
> Changing the default size like that is not advised because it requires going though the entire app,

@Al12rs @captainsandypants So what should I do about this?

When doing this, I did in fact go through every page of the App and overlay them over the design.

If what you say is true, for example, going with smaller icons in file tree(s) being a deliberate choice, then what should I use as the source of truth?

Do I need to take @captainsandypants and go through every screen again? But in huddle 

> Icons in various buttons were now different in size, e.g. advanced installer, GameWidget etc.

Because these, for example, were in fact matching the designs on Figma. And I even asked specifically about GameWidget since those had a different size.

File: src/Themes/NexusMods.Themes.NexusFluentDark/Styles/Controls/Button/SpineDownloadButtonStyles.axaml [hunk 1]
```diff
@@ -42,13 +42,18 @@
             <Setter Property="ClipToBounds" Value="False" />
         </Style>
 
+        <!-- Shrink on Click -->
+        <Style Selector="^:pressed">
+            <Setter Property="RenderTransform" Value="scale(0.92)" />
+        </Style>
```
[the remark this answers is not in the document]
Comment (Sewer56) [reply src/Themes/NexusMods.Themes.NexusFluentDark/Styles/Controls/Button/SpineDownloadButtonStyles.axaml:48]: 0f135be467b47d37454b1307554e3d82da04068c

**Comment by Al12rs** [comment]
Reviewing again and marking things that need fixing, not necessarily in this PR:
- [ ] GameWidged icon inconsistencies: 
image
- [ ] Tab headers icon size seems very big. Alignment definitely needs fixing: 
image
- [ ] some icons don't change color correctly: 
image
- [ ] Menu bar text and icon don't seem aligned: 
image
- [ ] TopBar settings and help icons seem too big: 
image

**Comment by Sewer56** [comment]
@Al12rs we'll probably need to do a review of the design in general with @captainsandypants at one point, and write up a list of things that need standardized. I was about to ask them for a huddle.

I noticed that even within the design, there were places where common icons had different sizes between the different designs.

**Comment by Sewer56** [comment]
> some icons don't change color correctly:

This one actually has to do with how the icon is loaded. I let Lausandy know about it last week ahead of time.

It'll be a bit more involved, so this is best left for a follow-up PR.

**Comment by Sewer56** [comment]
I opened

- https://github.com/Nexus-Mods/NexusMods.App/issues/1666

For the icon colouring. Shouldn't be too hard, I already did the research.

**Review by Al12rs — APPROVED** [review]

**Pull request merged**